spec/features/global_announcements_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe 'Global announcements', type: :feature do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    describe 'no global announcements' do
+      before do
+        instance.announcements.clear
```

Comment:
@lowjoel @wangqiang1208 , 

I've been following the conversation on both ends. I also agree with both sides - on Joel's end it is about correctness, right assertion, and code readability. On Wang Qiang's end, it is about abstracting repetitive code as much as possible. The hard part is how to choose which - especially in this situation when it gets a little grey. 

I learn towards the argument that readability is compromised if we abstract certain pre-conditions. To the experienced developer, either style should be fine. However, to the new RoR guy, reading tests with abstracted pre-conditions can be quite tough.  

If we consider each test to be an independent unit, then the specs should have the full, relevant set of clearly stated pre-conditions. *In this case, the relevant pre-conditions are mainly the models/controllers you are trying to test - usually announcements, and not the user authentication or account creation.*  

Wang Qiang, if you still not convinced maybe you can suggest a different way of writing the entire file and we can discuss?

----

Just for kicks, I ran `rspec spec/features/global_announcements_spec.rb` twice for both cases - results are below. Not the best method, but this was a quick and dirty way to measure the time taken.

##### Joel's suggestion (Slightly longer average of 0.04 seconds)
- `Finished in 0.62794 seconds (files took 1.86 seconds to load)`
- `Finished in 0.6446 seconds (files took 1.89 seconds to load)`

##### Wang Qiang's suggestion
- `Finished in 0.63257 seconds (files took 1.82 seconds to load)`
- `Finished in 0.63079 seconds (files took 1.84 seconds to load)`